File: minesweeper-java-qt/src/com/lattisi/minesweeperqt/Button.java
/*
 * Copyright (C) 2013 Tiziano Lattisi (http://www.lattisi.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.lattisi.minesweeperqt;

import com.trolltech.qt.core.QEvent;
import com.trolltech.qt.core.QObject;
import com.trolltech.qt.core.QSize;
import com.trolltech.qt.core.Qt;
import com.trolltech.qt.gui.QMouseEvent;
import com.trolltech.qt.gui.QSizePolicy;
import com.trolltech.qt.gui.QToolButton;

public class Button extends QToolButton {

    public final Signal0 rightClicked = new Signal0();

    public Button() {
        super();
        installEventFilter(this);
        setFixedSize(new QSize(36, 36));
        setIconSize(new QSize(32, 32));
        setSizePolicy(QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed);
    }

    @Override
    public boolean eventFilter(QObject object, QEvent event) {
        if( event.type() == QEvent.Type.MouseButtonPress ){
            if( ((QMouseEvent) event).button() == Qt.MouseButton.RightButton ){
                this.rightClicked.emit();
                return true;
            }
        }
        return super.eventFilter(object, event);

    }
}

File: minesweeper-python-qt/src/minesweeper/Button.py
from PyQt4 import QtGui, QtCore

class Button(QtGui.QToolButton):

    rightClicked = QtCore.pyqtSignal()

    def __init__(self):
        super(Button, self).__init__()
        self.installEventFilter(self)
        self.setFixedSize(36, 36)
        self.setIconSize(QtCore.QSize(32, 32))
        self.setSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)

    def eventFilter(self, obj, event):
        if event.type() == QtCore.QEvent.MouseButtonPress:
            if event.button() == QtCore.Qt.RightButton:
                self.rightClicked.emit()
                return True
        return super(Button, self).eventFilter(obj, event)
